### ISSUE TITLE
MixEnv false positive

### ISSUE THREAD

**Issue by fuelen** [rrrene/credo#907]
### Environment

* Credo version (`mix credo -v`): from master
* Erlang/Elixir version (`elixir -v`): 1.12
* Operating system: Linux

### What were you trying to do?
```
defmodule CredoTest do
  @myvar Mix.env() == :test

  def test do
    @myvar
  end
end
```

### Expected outcome
No warnings, calling Mix.env() in compile-time is safe.

### Actual outcome
```
┃ [W] ↗ There should be no calls to Mix.env in application code.
```

**Comment by rrrene** [comment]
Thx for reporting! :+1:

**Comment by rrrene** [comment]
Thanks for reporting this 😀  I am very sorry this took so long.

The issue is now fixed on `master`.

You can try this by setting the Credo dep to

```elixir
{:credo, github: "rrrene/credo"}
```

Please report back if your issue is solved! 👍

**Comment by fuelen** [comment]
@rrrene the issue is partially solved.
Newly generate Phoenix code has this code in Router
```elixir
  if Mix.env() in [:dev, :test] do
    import Phoenix.LiveDashboard.Router

    scope "/" do
      pipe_through :browser

      live_dashboard "/dashboard", metrics: HelloWeb.Telemetry
    end
  end
```
which produces warning as well.

**Comment by rrrene** [comment]
@fuelen You are right. Could you check again?

**Comment by fuelen** [comment]
@rrrene looks like the issue is solved :+1:

**Comment by rrrene** [comment]
This is now live in `v1.6.3`. Thx again!
